### PR TITLE
BFB-114: Implement Bullet Holes

### PR DESCRIPTION
**Pull request by unoctium1**
**JIRA Ticket**
https://perigongames.atlassian.net/browse/BFB-114

**Description**
Create bullet holes and particles with object pooling. Bullet holes are implemented via sprites.

**Screenshots / GIFs**
Bullet Holes

### CODE REVIEW

**Comment by unoctium1** [comment]
Let me know if the particle effects are too much at the moment, they might be a bit over the top and can be toned down if need be

File: Assets/_BForBoss/_Utility/Scripts/Extensions.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Perigon.Utility
+{
+    public static class Extensions
+    {
+        public static T GetRandomElement<T>(this IList<T> array)
+        {
+            return array[Random.Range(0, array.Count)];
```
Comment (m4rtint) [inline Assets/_BForBoss/_Utility/Scripts/Extensions.cs:11]: We actually have a utility withing RanomUtilitiy that get's a random element in an array

File: Assets/_BForBoss/_Weapons/Scripts/WallHitVFX.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        private const float MIN_ALPHA = 0f;
+        
+        [SerializeField] private Sprite[] _bulletHoleSprites;
+        private SpriteRenderer _spriteRenderer;
+        private ParticleSystem _particleSystem;
+
+        private ObjectPooler<WallHitVFX> _pool;
+
+        // Start is called before the first frame update
+        void Awake()
```
Comment (m4rtint) [inline Assets/_BForBoss/_Weapons/Scripts/WallHitVFX.cs:19]: set this as private

File: Assets/_BForBoss/_Weapons/Scripts/WallHitVFX.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        {
+            _spriteRenderer = GetComponent<SpriteRenderer>();
+            _particleSystem = GetComponentInChildren<ParticleSystem>();
+        }
+
+        public void Setup(ObjectPooler<WallHitVFX> pool)
```
Comment (m4rtint) [inline Assets/_BForBoss/_Weapons/Scripts/WallHitVFX.cs:25]: Can you rename this to `Initialize`

**Comment by m4rtint** [comment]
> Let me know if the particle effects are too much at the moment, they might be a bit over the top and can be toned down if need be

yeah can you tune it down lol. If you need any placeholder/reference weapon sparks, i have some if needed.

File: Assets/_BForBoss/_Utility/Scripts/Extensions.cs (hunk 1 above)
Comment (unoctium1) [reply Assets/_BForBoss/_Utility/Scripts/Extensions.cs:11]: Oh cool thanks, I checked IListExtensions but forgot about that one!

**Review by Gursaanj — APPROVED** [review]
Looks good to me

**Pull request merged**